### ISSUE TITLE
[Question] Communication between Workers / Browsers in the same Test

### ISSUE THREAD

**Issue by garciallorente-fer** [microsoft/playwright#9985]
My example: 
I am currently developing an automated test of a page which can only be accessed using a Link that needs to be generated previously in the same test, and can get invalid after submitting the form of that same page.

I want to run this test in multiple browsers, for which case I need to make sure that all 3 browsers access to the page before continuing with the submission of the form in only one of the 3 browsers, to be able to check afterwards with the other 2 that the link is indeed invalid after that.

Is there any good way of communicating between workers / browsers within the same test , to be able to synchronize the different workers / browsers inside this test? 

I need this test running in a Azure DevOps Pipeline as well.

**Comment by JoelEinbinder** [comment]
If I understand you correctly, you have a workflow like
1. create link
2. check link in firefox
3. check link in chromium
4. check link in webkit
5. submit link in chromium
6. check link is invalid in chromium
7. check link is invalid in firefox
8. check link is invalid in webkit

You need each of these steps to run in order because they all operate on the same link, but the browsers run in parallel so you can't.

Instead, run your test with one browser 3 times. Then you don't have to have a global lock on the link, because you have 3 unique links.

1. create link
2. check link in <browser>
3. submit link in <browser>
8. check link is invalid in <browser

**Comment by garciallorente-fer** [comment]
Thank you for your quick response Joel.

That proposed solution would work for my proposed example, although I forgot to mention that in my test I also want to use just one link for the 3 browsers, instead of 3 different links, because I want to check also that after one of the 3 browsers have submitted the form, the other 2 cannot submit the form and will get an error message. (as the link have been used) For that I would need to be already in the page.

Following your response, I suppose that I could do this by creating 3 test files as you proposed (each of them running in a different browser) and inside each of them I could create 2 pages, both opening the unique link. Then submitting in one the form and checking the validation after that in the other.
1. create link
2a. check link in page 1
2b. check link in page 2
3. submit form in page 1
4a. check invalid link page 1
4b. check invalid form page 2

I will do that , thank you for your quick response Joel.

Although I was wondering if there would be any way of communicating between browsers/workers to be able to do this in one file with one link, instead of 3 test files and 3 different links? or if there is any plans to implement something similar for the future?
Congratulations for the amazing job with Playwright :)

**Comment by garciallorente-fer** [comment]
I would also like to ask if there is any way to skip a test for some browsers without test.skip() , because test.skip() opens the instance of the browser but don't execute the test, but I would like to permanently execute that test only in one specific browser, without opening 3 instances and without having 3 reports of the same test.

**Comment by aslushnikov** [comment]
> Although I was wondering if there would be any way of communicating between browsers/workers to be able to do this in one file with one link, instead of 3 test files and 3 different links? or if there is any plans to implement something similar for the future?

@garciallorente-fer Nothing planned like this so far! 

> I would also like to ask if there is any way to skip a test for some browsers without test.skip() , because test.skip() opens the instance of the browser but don't execute the test, but I would like to permanently execute that test only in one specific browser, without opening 3 instances and without having 3 reports of the same test, and most importantly, without using more workers than needed.
Thanks in advance

This one is better to do using project configuration. You can collocate chrome-only or firefox-only tests and setup [`testIgnore`](https://playwright.dev/docs/api/class-testconfig#test-config-test-ignore) or [`testMatch`](https://playwright.dev/docs/api/class-testconfig#test-config-test-match) in the projects.

Feel free to file a new issue if there are more questions!
